rocAL/rocAL/source/pipeline/tensor.cpp

```diff
@@ -320,17 +374,39 @@ unsigned rocalTensor::copy_data(hipStream_t stream, void *host_memory, bool sync
 unsigned rocalTensor::copy_data(void *user_buffer) {
     if (_mem_handle == nullptr) return 0;
 
-#if ENABLE_HIP
-    if (_info._mem_type == RocalMemType::HIP) {
-        // copy from device to host
-        hipError_t status;
-        if ((status = hipMemcpyDtoH((void *)user_buffer, _mem_handle, _info.data_size())))
-            THROW("copy_data::hipMemcpyDtoH failed: " + TOSTR(status))
-    } else
-#endif
-    {
-        // copy from host to host
-        memcpy(user_buffer, _mem_handle, _info.data_size());
+    #if ENABLE_HIP
+        if (_info._mem_type == RocalMemType::HIP) {
+            // copy from device to device
+            hipError_t status;
+            if ((status = hipMemcpyDtoD((void *)user_buffer, _mem_handle, _info.data_size())))
+                THROW("copy_data::hipMemcpyDtoH failed: " + TOSTR(status))
+        } else if (_info._mem_type == RocalMemType::HOST) {
+            // copy from host to device
+            hipError_t status;
+            if ((status = hipMemcpyHtoD((void *)user_buffer, _mem_handle, _info.data_size())))
+                THROW("copy_data::hipMemcpyHtoD failed: " + TOSTR(status))
+        }
+    #else
+        if (_info._mem_type == RocalMemType::HOST)
+            memcpy(user_buffer, _mem_handle, _info.data_size());
+    #endif
+
+    return 0;
+}
```

Comment:
Do not make changes to copy_data existing API without ROI
It will be modified in another PR